apps/remix-ide/src/app/plugins/contractFlattener.tsx

```diff
@@ -1,52 +1,65 @@
 import React from 'react'
 import { Plugin } from '@remixproject/engine'
 import { customAction } from '@remixproject/plugin-api'
-import { concatSourceFiles, getDependencyGraph } from '@remix-ui/solidity-compiler'
+import { concatSourceFiles, getDependencyGraph, normalizeContractPath } from '@remix-ui/solidity-compiler'
 
 const _paq = window._paq = window._paq || []
 
 const profile = {
   name: 'contractflattener',
   displayName: 'Contract Flattener',
   description: 'Flatten solidity contracts',
-  methods: ['flattenAContract'],
+  methods: ['flattenAContract', 'flattenContract'],
   events: [],
+  maintainedBy: 'Remix',
 }
 
 export class ContractFlattener extends Plugin {
-  fileName: string
   constructor() {
     super(profile)
-    this.fileName = ''
   }
 
   onActivation(): void {
     this.on('solidity', 'compilationFinished', async (file, source, languageVersion, data, input, version) => {
-      await this.flattenContract(source, this.fileName, data)
+      await this.flattenContract(source, file, data)
     })  
     _paq.push(['trackEvent', 'plugin', 'activated', 'contractFlattener'])
   }
 
   async flattenAContract(action: customAction) {
-    this.fileName = action.path[0]
-    await this.call('solidity', 'compile', this.fileName)
+    this.call('manager', 'deactivatePlugin', 'solidityumlgen')
```

Comment:
Why do you deactivate 'solidityumlgen' here?

Reply:
This was from my trouble shooting. Will clean it up.